distribution/test/itests/test-itests-ddf-core/src/test/java/ddf/test/itests/core/DdfCoreIT.java

```diff
@@ -0,0 +1,477 @@
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.xml.xpath.XPathExpressionException;
+import org.apache.commons.io.FileUtils;
+import org.codice.ddf.itests.common.AbstractIntegrationTest;
+import org.codice.ddf.itests.common.config.UrlResourceReaderConfigurator;
+import org.codice.ddf.itests.common.csw.mock.FederatedCswMockServer;
+import org.codice.ddf.test.common.annotations.AfterExam;
+import org.codice.ddf.test.common.annotations.BeforeExam;
+import org.hamcrest.Matchers;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerSuite;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerSuite.class)
+public class DdfCoreIT extends AbstractIntegrationTest {
+
+  private static final DynamicUrl SECURE_ROOT_AND_PORT =
+      new DynamicUrl(DynamicUrl.SECURE_ROOT, HTTPS_PORT);
+
+  private static final DynamicUrl ADMIN_PATH =
+      new DynamicUrl(SECURE_ROOT_AND_PORT, "/admin/index.html");
+
+  private static final String RECORD_TITLE_1 = "myTitle";
+
+  private static final String RECORD_TITLE_2 = "myXmlTitle";
+
+  private static final DynamicPort RESTITO_STUB_SERVER_PORT = new DynamicPort(6);
+  private UrlResourceReaderConfigurator urlResourceReaderConfigurator;
+  private static StubServer server;
+  private static final String SAMPLE_DATA = "sample data";
+  private static FederatedCswMockServer cswServer;
+  private static final String DEFAULT_SAMPLE_PRODUCT_FILE_NAME = "sample.txt";
+  private static final String DEFAULT_URL_RESOURCE_READER_ROOT_RESOURCE_DIRS = "data/products";
+  private final List<String> resourcesToDelete = new ArrayList<>();
+
+  private static final int MAX_DOWNLOAD_RETRY_ATTEMPTS = 3;
+  private static final String CSW_STUB_SOURCE_ID = "cswStubServer";
+  private static final DynamicPort CSW_STUB_SERVER_PORT = new DynamicPort(7);
+  private static final DynamicUrl CSW_STUB_SERVER_PATH =
+      new DynamicUrl(INSECURE_ROOT, CSW_STUB_SERVER_PORT, "/services/csw");
+  private static final String POLL_INTERVAL = "pollInterval";
+  private static final int CSW_SOURCE_POLL_INTERVAL = 10;
+  private static final String CSW_SOURCE_WITH_METACARD_XML_ID = "cswSource2";
+  private static final String GMD_SOURCE_ID = "gmdSource";
+
+  @BeforeExam
+  public void beforeExam() throws Exception {
+    getCatalogBundle().setupMaxDownloadRetryAttempts(MAX_DOWNLOAD_RETRY_ATTEMPTS);
+
+    setupOpenSearch();
+
+    setupCswServer();
+
+    setupGmd();
+
+    getCatalogBundle().waitForFederatedSource(OPENSEARCH_SOURCE_ID);
```

Comment:
❓Are the tests actually using all these sources? It looks like the only one the tests really need is the OpenSearch source.

Reply:
Yes, Opensearch is the main one that is used, then CSW is used for a few. GMD is required by the CSW tests / source.